runtime-components/non-repudiation/src/main/scala/com/daml/nonrepudiation/KeyRepository.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation
+
+import java.security.{MessageDigest, PublicKey}
+
+import com.google.common.io.BaseEncoding
+
+import scala.collection.concurrent.TrieMap
+
+object KeyRepository {
+
+  trait Read {
+    def get(fingerprint: String): Option[PublicKey]
+  }
+
+  trait Write {
+    def put(key: PublicKey): String
+  }
+
+  private def hash(key: PublicKey): String = {
```

Comment:
we have the same fingerprint logic duplicated above. maybe worth reusing it?

Reply:
I'll have a look in the context of understanding how fingerprints are usually obtained for X.509 certificates and put it in the "core" library, if we decide to have one.

Reply:
Addressed in https://github.com/digital-asset/daml/pull/8608/commits/82cb80de118c91943e357689e17e588a2099b039